lib/internal/child_process.js

```diff
@@ -516,6 +529,16 @@ ChildProcess.prototype.kill = function(sig) {
   return false;
 };
 
+ChildProcess.prototype[SymbolAsyncDispose] = async function() {
+  if (!this.closed) {
+    const promise = EventEmitter.once(this, 'close');
```

Comment:
Kill happens asynchronously. Why are you doing this?

Reply:
since we want to wait for the close event, when the process and all its dependent streams are actually closed

Reply:
I still don't see why you need to create the promise before the call to kill?

Reply:
to avoid a race condition where `close` is emitted before I listen to it.

Reply:
That's not possible though? Kill will never emit close.

Reply:
got it.